#1: Importing Libraries and Loading Data

In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

# Inspect the first 5 rows
print("--- First 5 Rows ---")
display(df.head())

# Get basic information about the dataset (columns, data types, non-null counts)
print("\n--- Dataset Info ---")
df.info()

--- First 5 Rows ---


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI



--- Dataset Info ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    object 
 2   Customer          100 non-null    object 
 3   City              100 non-null    object 
 4   Region            100 non-null    object 
 5   Category          100 non-null    object 
 6   Product           100 non-null    object 
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12  Payment_Method    100 non-null    object 
dtypes: float64(2), int64(4), object(7)
memory usage: 10.3+ KB


#2: General Sales Statistics

In [2]:
# Calculate basic statistics
total_sales = df['Total_Sales'].sum()
average_sales = df['Total_Sales'].mean()
max_sales = df['Total_Sales'].max()
min_sales = df['Total_Sales'].min()
total_quantity = df['Quantity'].sum()
total_orders = df['Order_ID'].nunique()

# Print the statistics
print("--- Overall Sales Statistics ---")
print(f"Total Sales: ₹{total_sales:,.2f}")
print(f"Average Sales per Order: ₹{average_sales:,.2f}")
print(f"Highest Single Order Sales: ₹{max_sales:,.2f}")
print(f"Lowest Single Order Sales: ₹{min_sales:,.2f}")
print(f"Total Quantity Sold: {total_quantity} items")
print(f"Total Number of Orders: {total_orders}")

--- Overall Sales Statistics ---
Total Sales: ₹447,262.95
Average Sales per Order: ₹4,472.63
Highest Single Order Sales: ₹13,995.00
Lowest Single Order Sales: ₹449.10
Total Quantity Sold: 294 items
Total Number of Orders: 100


#3: Grouping and Aggregation (Category & City)

In [3]:
# 1. Total Sales by Category
category_sales = df.groupby('Category')['Total_Sales'].sum().sort_values(ascending=False).reset_index()
print("--- Total Sales by Category ---")
display(category_sales)

# 2. Total Sales by City (Top 5 Cities)
city_sales = df.groupby('City')['Total_Sales'].sum().sort_values(ascending=False).head(5).reset_index()
print("\n--- Top 5 Cities by Sales ---")
display(city_sales)

--- Total Sales by Category ---


,Category,Total_Sales
0,Sports,136026.45
1,Home & Kitchen,113329.90
2,Electronics,92086.00
3,Clothing,69028.20
4,Books,36792.40



--- Top 5 Cities by Sales ---


,City,Total_Sales
0,Hyderabad,59076.45
1,Chennai,47493.20
2,Lucknow,46194.65
3,Pune,41908.55
4,Mumbai,41457.80


#4: Grouping and Aggregation (Products & Payments)

In [4]:
# 1. Top 5 Best-Selling Products
top_products = df.groupby('Product')['Total_Sales'].sum().sort_values(ascending=False).head(5).reset_index()
print("--- Top 5 Best-Selling Products ---")
display(top_products)

# 2. Payment Method Popularity (Count of orders)
payment_methods = df['Payment_Method'].value_counts().reset_index()
payment_methods.columns = ['Payment_Method', 'Order_Count']
print("\n--- Payment Method Distribution ---")
display(payment_methods)

--- Top 5 Best-Selling Products ---


,Product,Total_Sales
0,Coffee Maker,51610.25
1,Jacket,47106.15
2,Cricket Bat,41483.40
3,Running Shoes,41285.25
4,Wireless Headphones,33877.40



--- Payment Method Distribution ---


,Payment_Method,Order_Count
0,UPI,26
1,Credit Card,22
2,Debit Card,20
3,Net Banking,17
4,Cash on Delivery,15


#5: Filtering and Sorting Data

In [5]:
# Filter: Identify orders where Total_Sales is greater than ₹10,000
high_value_orders = df[df['Total_Sales'] > 10000]
print(f"--- High Value Orders (> ₹10,000) : {len(high_value_orders)} orders ---")
display(high_value_orders[['Order_ID', 'Customer', 'Category', 'Total_Sales']])

# Sort: Top 5 Highest Individual Orders in the dataset
top_5_orders = df.sort_values(by='Total_Sales', ascending=False).head(5)
print("\n--- Top 5 Highest Individual Orders ---")
display(top_5_orders[['Order_ID', 'Customer', 'Product', 'Quantity', 'Total_Sales']])

--- High Value Orders (> ₹10,000) : 6 orders ---


,Order_ID,Customer,Category,Total_Sales
12,1013,Maryam Khan,Clothing,10620.75
47,1048,Sara Ahmed,Home & Kitchen,10497.00
48,1049,Ishita Gupta,Sports,13995.00
52,1053,Ananya Singh,Sports,13295.25
84,1085,Sana Malik,Home & Kitchen,12596.40
89,1090,Aditya Verma,Sports,12495.00



--- Top 5 Highest Individual Orders ---


,Order_ID,Customer,Product,Quantity,Total_Sales
48,1049,Ishita Gupta,Running Shoes,5,13995.00
52,1053,Ananya Singh,Running Shoes,5,13295.25
84,1085,Sana Malik,Coffee Maker,4,12596.40
89,1090,Aditya Verma,Cricket Bat,5,12495.00
12,1013,Maryam Khan,Jacket,5,10620.75


# **Key Observations and Findings**

Based on the execution of the analysis above, here are the meaningful observations extracted from the dataset:

*   **Overall Performance:** The store generated a total revenue of **₹447,262.95** from **100** unique orders, moving a total of **294** items. The average order value sits at **₹4,472.63**.
*   **Top Categories:** **Sports** is the highest-performing category (generating **₹136,026.45**), closely followed by **Home & Kitchen** (**₹113,329.90**). Books yielded the lowest revenue among the categories.
*   **Geographic Hotspots:** **Hyderabad** leads all cities in total sales volume (**₹59,076.45**), showing strong demand, with Chennai and Lucknow following in second and third place.
*   **Product Superstars:** The **Coffee Maker** is the absolute best-selling individual product by revenue (**₹51,610.25**), followed closely by the **Jacket** and the **Cricket Bat**.
*   **Payment Preferences:** **UPI** is the most favored payment method among customers (**26 transactions**), indicating a strong preference for digital wallets over traditional Net Banking or Cash on Delivery.
*   **Order Extremes:** The highest single transaction generated **₹13,995.00**, contrasting starkly with the minimum recorded order of just **₹449.10**.